packages/front-end/components/Settings/EditDataSource/DataSourceMetrics.tsx

```diff
@@ -244,27 +252,31 @@ export default function DataSourceMetrics({
                               </Tooltip>
                             ) : null}
                           </div>
-                          {editMetricsPermissions[metric.id] ? (
+                          {!metric.managedBy &&
```

Comment:
This change doesn't seem right.  Just because this metric is managed externally, doesn't mean you can't create a copy.

Reply:
Ah - I thought those were protected from duplicating. 

Do you think we should prevent a user from archiving an externally managed metric?

Reply:
Yes, I'm pretty sure archiving an official metric will fail in the back-end already, so we should block it in the UI as well

Reply:
Sounds good - just updated this logic to:

```ts
{editMetricsPermissions[metric.id].canDuplicate ||
                          (!metric.managedBy &&
                            editMetricsPermissions[metric.id].canUpdate) ? (
                            <MoreMenu className="px-2">
                              {editMetricsPermissions[metric.id]
                                .canDuplicate ? (
                                <button
                                  className="btn dropdown-item py-2"
                                  onClick={(e) => {
                                    e.stopPropagation();
                                    e.preventDefault();
                                    setModalData({
                                      current: {
                                        ...metric,
                                        managedBy: "",
                                        name: metric.name + " (copy)",
                                      },
                                      edit: false,
                                      duplicate: true,
                                    });
                                  }}
                                >
                                  <FaRegCopy /> Duplicate
                                </button>
                              ) : null}
                              {!metric.managedBy &&
                              editMetricsPermissions[metric.id].canUpdate ? (
                                <button
                                  className="btn dropdown-item py-2"
                                  color=""
                                  onClick={async () => {
                                    const newStatus =
                                      metric.status === "archived"
                                        ? "active"
                                        : "archived";
                                    await apiCall(`/metric/${metric.id}`, {
                                      method: "PUT",
                                      body: JSON.stringify({
                                        status: newStatus,
                                      }),
                                    });
                                    mutateDefinitions({});
                                    mutate();
                                  }}
                                >
                                  <FaArchive />{" "}
                                  {metric.status === "archived"
                                    ? "Unarchive"
                                    : "Archive"}
                                </button>
                              ) : null}
                            </MoreMenu>
                          ) : null}
```